installation/windows.md

```diff
@@ -129,3 +129,53 @@ To halt the Fluent Bit service, just execute the "stop" command.
 ```
 % sc.exe stop fluent-bit
 ``
+## Compile from Source
+
+If you need to create a custom exectable, you can use the following procedure to compile Fluent Bit by yourself.
+
+### Preparation
+
+First, you need Microsoft Visual C++ to compile Fluent Bit. You can install the minimum toolkit by the following command:
+
+```powershell
+PS> wget -o vs.exe https://aka.ms/vs/16/release/vs_buildtools.exe
+PS> start vs.exe
+```
+
+When asked which packages to install, choose "C++ Build Tools" and wait until the process finishes.
+
+Also you need to install flex and bison. One way to install them on Windows is to use [winflexbison](https://github.com/lexxmark/winflexbison).
+
+```powershell
+PS> wget -o winflexbison.zip https://github.com/lexxmark/winflexbison/releases/download/v2.5.22/win_flex_bison-2.5.22.zip
+PS> Expand-Archive winflexbison.zip -Destination C:\WinFlexBison
+PS> cp -Path C:\WinFlexBison\win_bison.exe C:\WinFlexBison\bison.exe
+PS> cp -Path C:\WinFlexBison\win_flex.exe C:\WinFlexBison\flex.exe
+PS> setx /M PATH "%PATH%;C:\WinFlexBison"
+```
+
+Also you need to install [git](https://git-scm.com/download/win) to pull the source code from the repository.
+
+### Compilation
+
+Open the start menu on Windows and type "Developer Command Prompt".
+
+Clone the source code of Fluent Bit.
+
+```
+% git clone https://github.com/fluent/fluent-bit
+% cd fluent-bit/build
+```
+
+Compile the source code.
+
+```
+% cmake ..
+% cmake --build .
+```
+
+Now you should be able to run Fluent Bit:
+
+```
+% .\bin\debug\fluent-bit.exe -i dummy -o stdout
+```
```

Comment:
so far so good.
Now when i run cpack it fails.

PS C:\Users\barni\fluent-bit\build> & 'C:\Program Files\CMake\bin\cpack.exe' -G ZIP
CPack: Create package using ZIP
CPack: Install projects
CPack: - Install project: fluent-bit []
CMake Error at C:/Users/barni/fluent-bit/build/src/cmake_install.cmake:39 (file):
  file INSTALL cannot find
  "C:/Users/barni/fluent-bit/build/bin/Release/fluent-bit.dll": File exists.
Call Stack (most recent call first):
  C:/Users/barni/fluent-bit/build/cmake_install.cmake:44 (include)


CMake Error at C:/Users/barni/fluent-bit/build/src/cmake_install.cmake:47 (file):
  file INSTALL cannot find
  "C:/Users/barni/fluent-bit/build/bin/Release/fluent-bit.exe": File exists.
Call Stack (most recent call first):
  C:/Users/barni/fluent-bit/build/cmake_install.cmake:44 (include)


CMake Error at C:/Users/barni/fluent-bit/build/src/cmake_install.cmake:55 (file):
  file INSTALL cannot find
  "C:/Users/barni/fluent-bit/build/bin/Release/fluent-bit.pdb": File exists.
Call Stack (most recent call first):
  C:/Users/barni/fluent-bit/build/cmake_install.cmake:44 (include)

Reply:
Try `cpack -G ZIP -C Debug` and see if  it works.